### PR TITLE
Fix retain cycle

### PR DESCRIPTION
**Pull request by lorraine-hatch**
This is a fix that I implemented to address this issue that I opened: https://github.com/apollographql/apollo-ios/issues/2437

### CODE REVIEW

File: Sources/Apollo/ApolloStore.swift [hunk 1]
```diff
@@ -218,6 +213,12 @@ public class ApolloStore {
       accumulator: Accumulator
     ) throws -> Accumulator.FinalResult {
       let object = try loadObject(forKey: key).get()
+      
+      let executor = GraphQLExecutor { object, info in
+        return object[info.cacheKeyForField]
+      } resolveReference: { reference in
+        self.loadObject(forKey: reference.key)
+      }
```
Comment (lorraine-hatch) [inline Sources/Apollo/ApolloStore.swift:222]: This is how it was in your main branch. That's where I got the idea to try reverting it. Cleans up the retain cycles 🙌

**Comment by AnthonyMDev** [comment]
Thanks so much for looking into this! Your PR definitely identified where the retain cycle was, which was incredibly helpful and saved us a ton of time. I appreciate it!

In https://github.com/apollographql/apollo-ios/pull/2476, I've gone ahead with an approach to this that uses a `[weak self]` capture list because it should be more performant (keeping a single `executor` instead of creating a new one for each read in the transaction). I also added a unit test to make sure that this fixes the memory leak and we don't let it regress and come back in the future. I'll be merging that PR in shortly.

Once it's merged, if you could point your codebase at the `release/1.0` branch and verify that the fix does indeed remove the retain cycle you are experiencing, that would be great!

**Comment by lorraine-hatch** [comment]
@AnthonyMDev great! I will try it out later today and let you know.

**Comment by lorraine-hatch** [comment]
@AnthonyMDev I've pulled in the `release/1.0` branch, but I'm running into build errors in the generated code: 

CleanShot 2022-08-31 at 10 45 40@2x

Any ideas what's happening here?

**Comment by AnthonyMDev** [comment]
The API for the generated code has changed a bit. Please re-run codegen! You'll also probably need to remove the `Schema.swift` and Schema+CacheKeyResolution.swift` files from your project.

More info about this can be found in the [Beta 3 release notes](https://github.com/apollographql/apollo-ios/releases/tag/1.0.0-beta.3)!

**Comment by lorraine-hatch** [comment]
oh awesome, thank you!

**Comment by lorraine-hatch** [comment]
Ok it worked 🥳, no more retain cycle!

**Comment by lorraine-hatch** [comment]
Does the latest beta release include this?

**Comment by AnthonyMDev** [comment]
No, we’ll get this into beta 4!

**Comment by lorraine-hatch** [comment]
Sounds good! Is there an anticipated release date?

**Comment by calvincestari** [comment]
I think we can tentatively schedule a beta 4 release for the end of the week; I'll confirm with @AnthonyMDev later today. Here's the current [diff](https://github.com/apollographql/apollo-ios/compare/1.0.0-beta.3...release/1.0) of what would be included with a beta 4 release.

**Comment by lorraine-hatch** [comment]
Sounds great, thanks @calvincestari!

**Pull request closed without merging**